src/Util/ComposeFileManipulator.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ * This file is part of the Symfony MakerBundle package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\MakerBundle\Util;
+
+/**
+ * Manipulate Docker Compose Files.
+ *
+ * @author Jesse Rushlow <jr@rushlow.dev>
+ *
+ * @internal
+ * @final
+ */
+class ComposeFileManipulator
+{
+    public const COMPOSE_FILE_VERSION = '3.7';
+
+    private $composeData;
+
+    public function __construct(string $contents)
+    {
+        if ('' === $contents) {
+            $this->composeData = $this->getBasicStructure();
+        } else {
+            $this->composeData = (new YamlSourceManipulator($contents))->getData();
```

Comment:
Per my comment about using `YamlSourceManipulator` to actually make changes to the YAML, we should probably maintain an instance of `YamlSourceManipulator` on a private property, then use `->getData()` on it whenever we need to look at data. And, all mutations would go through the YSM instead of on a local `composeData` array (which would now not exist).

Reply:
done